From af479865cabf655abefc7424677e8ec67cd02688 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:56 -0400
Subject: [tools] perf: Add support for the xtensa architecture

Message-id: <1408953210-26343-289-git-send-email-jolsa@redhat.com>
Patchwork-id: 89002
O-Subject: [PATCH RHEL7.1 BZ1133083 288/322] perf tools: Add support for the xtensa architecture
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 3a46817f674389d5fb2f4dce19fbc73ebd1976bb
Author: Baruch Siach <baruch@tkos.co.il>
Date: Mon Jan 13 12:27:35 2014 +0200

description
===========
Tested using kernel tracepoints on a QEMU simulated environment.

Kernel support for perf depends on the patch "xtensa: enable
HAVE_PERF_EVENTS", which is scheduled for v3.14.

Hardware performance counters are not supported under xtensa yet.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/perf.h b/tools/perf/perf.h
index 3c2f213..7daa806 100644
--- a/tools/perf/perf.h
+++ b/tools/perf/perf.h
@@ -132,6 +132,13 @@
 #define CPUINFO_PROC "CPU"
 #endif
 
+#ifdef __xtensa__
+#define mb()  asm volatile("memw" ::: "memory")
+#define wmb()  asm volatile("memw" ::: "memory")
+#define rmb()  asm volatile("" ::: "memory")
+#define CPUINFO_PROC "core ID"
+#endif
+
 #define barrier() asm volatile ("" ::: "memory")
 
 #ifndef cpu_relax
-- 
1.7.1